tools/android/packaging/xbmc/src/ExternalPermission.java.in

```diff
@@ -0,0 +1,82 @@
+package @APP_PACKAGE@;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.os.Bundle;
+
+public class ExternalPermission extends Activity
+{
+    public static final String PERMISSION_ACTION = "@APP_PACKAGE@.permission.ExternalFiles";
+
+    private SharedPreferences mPreferences;
+
+    private void acceptPackage(String pkgName) {
+        mPreferences.edit().putBoolean(pkgName, true).commit();
+    }
+
+    private void denyPackage(String pkgName) {
+        mPreferences.edit().putBoolean(pkgName, false).commit();
+    }
+
+    private boolean newPackage(String pkgName) {
+        return !mPreferences.contains(pkgName);
+    }
+
+    private boolean acceptedPackage(String pkgName) {
+        return mPreferences.getBoolean(pkgName, false);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mPreferences = getSharedPreferences("external_permissions", 0);
+        final String requester = getCallingPackage();
+        if (getIntent() != null && PERMISSION_ACTION.equals(getIntent().getAction()) && requester != null) {
+            if (newPackage(requester)) {
+                setContentView(R.layout.activity_splash);
+                AlertDialog dialog = new AlertDialog.Builder(this)
+                    .setCancelable(false)
+                    .setTitle("WARNING")
+                    .setMessage(requester+" requested access to @APP_NAME@'s external settings directory. If you approve this request, "+requester+
+                        " will have permission to change your configurations, or even read stored passwords for your media sources."+
+                        "\n Only accept the request if you really trust that application!")
```

Comment:
Suggestion: `\n O` -> `\nO`

Does this string need to be translated?

Reply:
None of the strings in the Android sources are translated so I didn't plan to introduce translation with this PR.